### ISSUE TITLE
Username and Password leaked to logs

### ISSUE THREAD

**Issue by denniseigner** [pycontribs/jira#960]
**Describe the bug**
When using username + password authentication, those credentials are leaked into logs when the jira server cannot be reached.

**To Reproduce**
```
from jira import JIRA

server = 'https://jira.foo.bar/'
auth = ('user', 'supersecretpassword')

jira = JIRA(server, auth=auth)
```

Outputs:

```
WARNING:root:HTTPSConnectionPool(host='jira.foo.bar', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7fb2e0cb4d30>: Failed to establish a new connection: [Errno -2] Name or service not known',)) while doing POST https://jira.foo.bar/rest/auth/1/session [{'data': '{"username": "user", "password": "supersecretpassword"}', 'headers': {'User-Agent': 'python-requests/2.24.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'application/json,*.*;q=0.9', 'Connection': 'keep-alive', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'X-Atlassian-Token': 'no-check'}}]
```

**Expected behavior**
Either do not print the `data` field to the logs or replace password field with stars.

**Version Information**
Python Interpreter: 3.6.9
jira-python: 3.0
OS: Ubuntu 18.04
IPython (Optional): <VERSION>
Other Dependencies:

**Comment by Flupp** [comment]
Although I cannot reproduce the problem with version 3.0, I had the same problem with version 2.0.0. Some time ago, I already proposed a fix in pull request #928.

**Comment by Flupp** [comment]
Btw, the problem can be worked around by adapting the log level:

```py
import logging

logging.basicConfig(level='ERROR')
```
